airflow/www/security.py

```diff
@@ -195,6 +195,19 @@ def init_role(self, role_name, role_vms, role_perms):
         else:
             self.log.info('Existing permissions for the role:%s within the database will persist.', role_name)
 
+    def delete_role(self, role_name):
```

Comment:
Have you checked and see if FAB's SecurityManager provide API to delete role?

Reply:
And maybe we need to check and make sure that user shouldn't delete those 5 default roles defined in the security.py.

Reply:
Yeah, it doesn't appear to. These are the methods it does have:

```
    """
    ----------------------
     PRIMITIVES FOR ROLES
    ----------------------
    """
    def find_role(self, name):
        raise NotImplementedError

    def add_role(self, name):
        raise NotImplementedError

    def get_all_roles(self):
        raise NotImplementedError
```

From https://flask-appbuilder.readthedocs.io/en/latest/_modules/flask_appbuilder/security/manager.html